### ISSUE TITLE
Failing to read XLS file: process doesn't terminate.

### ISSUE THREAD

**Issue by Yv-vY** [zdavatz/spreadsheet#258]
Using the spreadsheet gem I've hit this bug when trying to load the attached XLS file, which obviously is bloated following a "save as" from XLSX to XLS in Microsoft Excel.
When selecting the worksheet 0, the ruby process starts to have high CPU usage and never completes (at least in 5 minutes).

Notes:
- there's only ~ 300 lines of data in the file
- the file reads perfectly with LibreOffice too
[BloatedExcelFile.zip](https://github.com/zdavatz/spreadsheet/files/5889045/BloatedExcelFile.zip)

Thanks for your support.

**Comment by zdavatz** [comment]
So the file was created with Microsoft Office and saved as XLSX? Then it was saved with Microsoft Office as XLS?

**Comment by Yv-vY** [comment]
Yes this is correct. The only thing is that it has been first created with Microsoft Excel and saved (to default XLSX format) and closed then opened again and saved as XLS.

**Comment by zdavatz** [comment]
What happens if you save the file to XLS using LibreOffice? Does spreadsheet gem then work?

**Comment by zdavatz** [comment]
And _**why**_ was was the file first saved as XLSX and not as XLS?

**Comment by Yv-vY** [comment]
> What happens if you save the file to XLS using LibreOffice? Does spreadsheet gem then work?
Yes the file shrinks to 97 KB and then everything works fine

> And why was was the file first saved as XLSX and not as XLS?
Because the user didn't pay attention to this when he created the file (XLSX is the default)

My concern is not about the facts there are some limitation with the spreadsheet gem, that's fine with me. But at least it should fail nicely if it can't process some XLS files. Else it's a nightmare for users of any system making use of this gem (we can't possibly tell them to check for bloated files and/or expect them to guess that if they can't import/process some data in your system it's because the file is bloated).

**Comment by zdavatz** [comment]
I kind of a agree and if you have an idea how to "fail nicely" if spreadsheet gem can not process "an XLS" file then please submit a patch with the according test cases.

I am happy that LibreOffice is doing a good job.
